### DISCUSSION TITLE
Underscore in domain name

### DISCUSSION THREAD

**Discussion by aubreythreadgill** [plausible/analytics#3581] [Anything Else]
Hi @metmarkosaric, I noted a similar discussion #2122 last year, and it looks like you've added additional special characters over time.

Do you have any plans to support underscores in domain names?

I'm using Plausible in the white-label approach of offering end users their own analytics dashboards. The username format for my web app is like that of X/Twitter or Instagram (e.g. x.​com/alice_green).

Is my understanding correct that I would not be able to provision a Plausible `site` for that example user? (Or perhaps even the stats of two separate users `tapistr.ee/AliceGreen` and `tapistr.ee/alice_green` would be combined?)

Thanks as always for your help on here!

![Screenshot 2023-12-01 at 9 23 00 AM](https://github.com/plausible/analytics/assets/94575293/48eb13eb-688c-4ba9-a73d-337fadf1ca8f)

Best,
Aubrey

**Comment by metmarkosaric** [comment]
Hi @aubreythreadgill! The domain name you add to Plausible doesn't necessarily need to match the actual domain name where you insert the snippet. So for your example in the screenshot you could use a dash instead of underscore or something like "iloveplausible"

**Reply by aubreythreadgill** [reply]
Aha! I don't allow hyphens in usernames, so I'll just replace the underscores with hyphens in the plausible snippet to avoid any duplicates. Thank you Marko! I thought you might have a simple solution.

```
let username = "alice_green"
const plausibleSiteName = username.replace(/_/g, '-'); //replace underscores with hyphens

 <script
              defer
              data-domain={`tapistr.ee/${plausibleSiteName}`}
              src="https://plausible.io/js/script.outbound-links.exclusions.js"
              data-exclude="/i/analytics"
            ></script>
            ```
